model/convert/service_event.go

```diff
@@ -589,9 +589,14 @@ func convertParticipants(cdcParticipants []cadence.Value) (flow.IdentityList, er
 		}
 
 		identity := &flow.Identity{
-			Address: string(address),
-			Weight:  uint64(initialWeight),
-			Role:    flow.Role(role),
+			IdentitySkeleton: flow.IdentitySkeleton{
+				InitialWeight: uint64(initialWeight),
+				Address:       string(address),
+				Role:          flow.Role(role),
+			},
+			DynamicIdentity: flow.DynamicIdentity{
+				Weight: uint64(initialWeight),
```

Comment:
Related to previous comment, we should:
- more precisely define `initialWeight` (from registration, or from beginning of current epoch)
- consider either:
  - explicitly providing `weight` (separately from `initialWeight`) in the service event, or
  - modify `EpochSetup` event to contain only `IdentitySkeleton`s (no dynamic weight)

Reply:
> modify `EpochSetup` event to contain only `IdentitySkeletons` (no dynamic weight)

+1

Reply:
Captured in https://github.com/onflow/flow-go/issues/4649#issue-1860381016